mne/io/nsx/tests/test_nsx.py

```diff
@@ -0,0 +1,135 @@
+# Author: Proloy Das <pdas6@mgh.harvard.edu>
+#
+# License: BSD-3-Clause
+import os
+import numpy as np
+import pytest
+
+from mne.io import read_raw_nsx
+from mne.datasets.testing import data_path, requires_testing_data
+from mne.io.constants import FIFF
+
+
+testing_path = data_path(download=False)
+nsx_21_fname = os.path.join(testing_path, "nsx", "test_NEURALSG_raw.ns3")
+nsx_22_fname = os.path.join(testing_path, "nsx", "test_NEURALCD_raw.ns3")
+nsx_31_fname = os.path.join(testing_path, "nsx", "test_BRSMPGRP_raw.ns3")
+
+
+@requires_testing_data
+def test_nsx_ver_31():
+    """Primary tests for BRSMPGRP reader."""
+    raw = read_raw_nsx(nsx_31_fname)
+    assert getattr(raw, "_data", False) is False
+    assert raw.info["sfreq"] == 2000
+
+    # Check info object
+    assert raw.info["meas_date"].day == 31
+    assert raw.info["meas_date"].year == 2023
+    assert raw.info["meas_date"].month == 1
+    assert raw.info["chs"][0]["cal"] == 0.6103515625
+    assert raw.info["chs"][0]["range"] == 0.001
+
+    # check raw_extras
+    for r in raw._raw_extras:
+        assert r["orig_format"] == raw.orig_format
+        assert r["orig_nchan"] == 128
+        assert len(r["timestamp"]) == len(r["nb_data_points"])
+        assert len(r["timestamp"]) == len(r["offset_to_data_block"])
+
+    # Check annotations
+    assert raw.annotations[0]["onset"] * raw.info["sfreq"] == 101
+    assert raw.annotations[0]["duration"] * raw.info["sfreq"] == 49
+
+    raw.load_data()
```

Comment:
For this I would `@pytest.mark.parametrize('preload', (True, False))` and then never call this `load_data()` here

Reply:
Thanks for the insight. This is completely removed now.